Sources/MapboxMaps/Style/Style.swift

```diff
@@ -291,44 +234,47 @@ public class Style {
         }
     }
 
+    // MARK: Light
+
+    /// Gets the value of a style light property.
+    ///
+    /// - Parameter property: Style light property name.
+    ///
+    /// - Returns: Style light property value.
+    public func lightProperty(_ property: String) -> Any {
+        return _lightProperty(property).value
+    }
+
     // MARK: Terrain
 
     /// Sets a terrain on the style
+    ///
     /// - Parameter terrain: The `Terrain` that should be rendered
-    /// - Returns: Result type with `.success` if terrain is successfully applied. `TerrainError` otherwise.
-    @discardableResult
-    public func setTerrain(_ terrain: Terrain) -> Result<Bool, TerrainError> {
-        do {
-            let terrainData = try JSONEncoder().encode(terrain)
-            let terrainDictionary = try JSONSerialization.jsonObject(with: terrainData)
-            let expectation = styleManager.setStyleTerrainForProperties(terrainDictionary)
-
-            return expectation.isValue() ? .success(true)
-                                         : .failure(.addTerrainFailed(expectation.error as? String))
-        } catch {
-            return .failure(.decodingTerrainFailed(error))
+    ///
+    /// - Throws:
+    ///     An error describing why the operation was unsuccessful.
+    public func setTerrain(_ terrain: Terrain) throws {
```

Comment:
I think we decided to make terrain a computed property with an explict setter / getter (and fatally erroring if unsuccessful). Same with Light.

Reply:
Yep, will address in the next PR :)